apollo/agent/serde.py

```diff
@@ -0,0 +1,35 @@
+import json
+from datetime import (
+    date,
+    datetime,
+)
+from typing import Any
+
+from apollo.agent.constants import (
+    ATTRIBUTE_NAME_DATA,
+    ATTRIBUTE_NAME_TYPE,
+    ATTRIBUTE_VALUE_TYPE_DATE,
+    ATTRIBUTE_VALUE_TYPE_DATETIME,
+)
+
+
+class AgentSerializer(json.JSONEncoder):
+    @classmethod
+    def serialize(cls, value: Any) -> Any:
+        if isinstance(value, datetime):
+            return {
+                ATTRIBUTE_NAME_TYPE: ATTRIBUTE_VALUE_TYPE_DATETIME,
+                ATTRIBUTE_NAME_DATA: value.isoformat(),
+            }
+        elif isinstance(value, date):
+            return {
+                ATTRIBUTE_NAME_TYPE: ATTRIBUTE_VALUE_TYPE_DATE,
+                ATTRIBUTE_NAME_DATA: value.isoformat(),
+            }
+        return value
+
+    def default(self, obj: Any):
+        serialized = self.serialize(obj)
+        if serialized is not obj:  # no serialization happened
```

Comment:
shouldn't the comment here be " # serialization happened" ?

Reply:
yup good catch